### PR TITLE
Alternate site

### PR DESCRIPTION
**Pull request by jzuech3**
We'd like the ability to map a site to another site (e.g. ifsta-alpha.nextthought.com to ifsta.nextthought.com). This would simplify clients that would need a complicated site hierachy tree

### CODE REVIEW

File: src/nti/site/_compat.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
```
Comment (jamadden) [inline src/nti/site/_compat.py]: Please don't add this file. It's only used in one place, and that one place doesn't need it.

File: src/nti/site/zcml.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    source_site_name = TextLine(title=u"The source site name")
+
+    target_site_name = TextLine(title=u"The target site name")
+
+
+def registerSiteMapping(_context, source_site_name, target_site_name):
+    """
+    Create and register a site mapping, as a utility under the `source_site_name`.
+    """
+    site_mapping = SiteMapping( source_site_name=text_(source_site_name),
```
Comment (jamadden) [inline src/nti/site/zcml.py]: The call to `text_` should not be needed because the schema should have already done that.

Comment (jamadden) [inline src/nti/site/zcml.py]: Please fix the spacing here (no space after opening bracket, align the lines).

File: src/nti/site/interfaces.py [hunk 3]
```diff
@@ -124,3 +139,21 @@ def __call__(func, retries=0, sleep=None, site_names=(), side_effect_free=False,
 
         :return: The value returned by the first successful invocation of `func`.
         """
+
+class ISiteMapping(interface.Interface):
+    """
+    Maps a site name to an alternate site. Useful when we do not want full
+    fledged sites in the dataserver. Should only be used after checking for an
```
Comment (jamadden) [inline src/nti/site/interfaces.py]: We should try to avoid specific language about "dataserver" in favor of more generic descriptions.

**Comment by jamadden** [comment]
Instead of merging master into a branch (which we will then merge into master, creating a confusing history) can you (a) delete that commit and (b) rebase this branch on master?

**Comment by jamadden** [comment]
I have it rebased locally. I can push that if you'd like.

**Comment by jamadden** [comment]
Rebase pushed, per request. Still need the test and the changenote.

**Comment by jamadden** [comment]
I rebased again to remove redundant commits " 7ea8658 Rebase" and " fc61e99 Merge branch 'alternate_site' of https://github.com/NextThought/nti.site into alternate_site"

**Comment by jamadden** [comment]
@jzuech3 Are you ready for a PyPI release?

**Comment by jzuech3** [comment]
Yes please. We probably only need the release prior to Thursday. Thanks @jamadden

**Comment by jamadden** [comment]
1.2.0 has been pushed to PyPI, courtesy of [zest.releaser's](http://zestreleaser.readthedocs.io/en/latest/) `fullrelease` command. Let me know if you have a PyPI username you'd like to have access to this project.

**Pull request merged**